### PR TITLE
Added ManagedIdentity authentication support for azure-cosmos-spark in Databricks

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
This PR allows using Managed Identity authentication in Databricks and HDInsights. In both of the spark environments a user provided Managed Identity is used as the workspace's managed identity. This PR allows using this workspace managed identity to authenticate against Cosmos DB.

```scala
val cfg = Map("spark.cosmos.accountEndpoint" -> cosmosEndpoint,
    "spark.cosmos.auth.type" -> "ManagedIdentity",
    "spark.cosmos.account.subscriptionId" -> subscriptionId,
    "spark.cosmos.account.tenantId" -> tenantId,
    "spark.cosmos.account.resourceGroupName" -> resourceGroupName,
    "spark.cosmos.database" -> cosmosDatabaseName,
    "spark.cosmos.container" -> cosmosContainerName,
)
```

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by FabianMeiswinkel** [comment]
Add config description

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/CosmosConfigSpec.scala [hunk 1]
```diff
@@ -64,7 +64,7 @@ class CosmosConfigSpec extends UnitSpec with BasicLoggingTrait {
 
           endpointConfig.endpoint shouldEqual sampleProdEndpoint
 
-          val aadAuthConfig = endpointConfig.authConfig.asInstanceOf[CosmosAadAuthConfig]
+          val aadAuthConfig = endpointConfig.authConfig.asInstanceOf[CosmosServicePrincipalAuthConfig]
```
Comment (tvaron3) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/CosmosConfigSpec.scala:67]: Should this variable be renamed servicePrincipleAuthConfig as well ?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/CosmosConfigSpec.scala:67]: Done

**Comment by FabianMeiswinkel** [comment]
> Add config description

Done

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala [hunk 2]
```diff
@@ -594,10 +605,13 @@ private object AzureEnvironmentType extends Enumeration {
 trait CosmosAuthConfig {}
 
 private case class CosmosMasterKeyAuthConfig(accountKey: String) extends CosmosAuthConfig
-private case class CosmosAadAuthConfig(
+private case class CosmosServicePrincipalAuthConfig(
                                        clientId: String,
                                        tenantId: String,
                                        clientSecret: String) extends CosmosAuthConfig
+private case class CosmosManagedIdentityAuthConfig( tenantId: String,
+                                                     clientId: Option[String],
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala:613]: should this be required? (same question for resourceId)

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala [hunk 3]
```diff
@@ -614,7 +628,8 @@ private object CosmosAuthConfig {
         parseFromStringFunction = authTypeAsString =>
             CosmosConfigEntry.parseEnumeration(authTypeAsString, CosmosAuthType),
         helpMessage = "There are two auth types are supported currently: " +
-            "`MasterKey`(PrimaryReadWriteKeys, SecondReadWriteKeys, PrimaryReadOnlyKeys, SecondReadWriteKeys), `ServicePrincipal`")
+            "`MasterKey`(PrimaryReadWriteKeys, SecondReadWriteKeys, PrimaryReadOnlyKeys, SecondReadWriteKeys), " +
+            "`ServicePrincipal` and 'ManagedIdentity' (when the underlying Spark runtime supports it)")
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala:632]: which spark runtime does not support it?

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala [hunk 4]
```diff
@@ -594,10 +605,13 @@ private object AzureEnvironmentType extends Enumeration {
...
 
 private case class CosmosMasterKeyAuthConfig(accountKey: String) extends CosmosAuthConfig
-private case class CosmosAadAuthConfig(
+private case class CosmosServicePrincipalAuthConfig(
                                        clientId: String,
                                        tenantId: String,
                                        clientSecret: String) extends CosmosAuthConfig
+private case class CosmosManagedIdentityAuthConfig( tenantId: String,
+                                                     clientId: Option[String],
+                                                     resourceId: Option[String]) extends CosmosAuthConfig
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala:614]: should resourceId be required?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/AccountDataResolver.scala [hunk 5]
```diff
@@ -3,6 +3,13 @@
 
 package com.azure.cosmos.spark
 
+import reactor.core.publisher.Mono
```
[the remark this answers is not in the document]
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/AccountDataResolver.scala]: Fixed

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala (hunk 3 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala]: Fixed

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/AccountDataResolver.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/AccountDataResolver.scala]: Fixed

**Pull request merged**